Readme.md

```diff
@@ -129,13 +129,276 @@ with keys corresponding to the options described in [CLI Arguments](#cliargs):
+  log line. It looks for fields `name`, `pid`, and `hostname` and returns a formatted
+  string using the fields it finds.
+* `semicolon` (*builder*) - Appends a semicolon to the output if values have already been
+  added to the output.
+* `message` (*parser*, *builder*) - Prettifies a message string if the given `input` object
+  has a `message` property.
+* `eol` (*builder*) - Adds an end-of-line to the output. Uses `\r\n` if the `crlf` option
+  is `true` or `\n` otherwise. Defaults to `\n`.
+* `error` (*builder*) - Given a log object that has a `type: 'Error'` key, prettifies the
+  object and returns the result.
+* `object` (*builder*) - Prettifies a standard object. Special care is taken when
+  processing the object to handle child objects that are attached to keys known to
+  contain error objects.
+
+The built-in log processing sequence can be replaced with a custom sequence comprised
+of a mix of built-in and custom log processors. To customize the log processing
+sequence, pass an array of log processors to the `processors` option. They will be
+executed in the order in which they are specified.
+
+To use a built-in log processor in the custom sequence, simply add a string specifying
+its name to the `processors` array. Any of the log processors in the list above can be
+used, except for the `json` log processor.
+
+Do not add the built-in `json` log processor to the custom sequence. The `json` log
+processor is always executed first, even when the `processors` option is specified. It
+cannot be removed or replaced. This is intended to ensure that any subsequent log
+processors in the sequence will always operate on an object consistently parsed from the
+original `ndjson` input.
+
+The default sequence is equivalent to specifying the `processors` option as follows:
+```js
+{
+  processors: [
+    // 'json', - the `json` log processor is always implicitly added at the beginning
+    'primitives',
+    'search',
+    'ignore',
+    'time',
+    'level',
+    'metadata',
+    'semicolon',
+    'message',
+    'eol',
+    'error',
+    'object'
+  ]
+}
+```
+
+By changing the order of the built-in log processors or excluding some of them, the
+output can changed. For example, to only show the time, level and actual message,
+while reversing the order of the time and level, the sequence can be specified as
+follows: `{ processors: ['level', 'time', 'message'] }`.
+
+Custom log processors can also be added to the log processing sequence. To add a
+custom log processor, use one of the following structures:
+
+* **parser** object
+  ```js
+  {
+    parse(input, context) {}
+  }
+  ```
+* **builder** object
+  ```js
+  {
+    build(lineParts, context) {}
+  }
+  ```
+* complete **parser** and **builder** object
+  ```js
+  {
+    parse(input, context) {}
+    build(lineParts, context) {}
+  }
+  ```
+* **parser** function
+  ```js
+  (input, context) => {}
+  ```
```

Comment:
Now that you mention it, I agree.

I left it in because it was really easy to, but it now occurs to me that there may be another reason to get rid of the parser function: it creates an inconsistency with the builders because matching support for a builder function doesn't exist and can't easily be added.

Unless someone can come up with a really good reason to leave it in, I'll take it out.